Feature/cleanup dashboard system basics

mnakalay opened the pull request:
This PR takes care of #8557. Details are listed on the issue page.

It requires a companion Bedrock PR https://github.com/concrete5/bedrock/pull/50

Things that were done:

- replaced all .btn-default with .btn-secondary
- replaced all .pull-left and .pull-right with .float-left and .float-right
- changed some markup on single pages
- made sure success messages were set as "success" and not "message"
- fixed icons for text and date/time attributes (FontAwesome changed their names and the polyfill was not working)
- modified the getListIconElement() method of the FontAwesomeIconFormatter class to make all icons have the same width using the class name .fa-fw
- Made the CKeditor plugin preview fancier with a "loading" text and spinner and a debounce function when adding extra plugins

## Pages that were modified ##

/dashboard/system/basics/name
/dashboard/system/basics/attributes
/dashboard/system/basics/attributes/edit
/dashboard/system/basics/accessibility
/dashboard/system/basics/social
/dashboard/system/basics/social/edit
/dashboard/system/basics/icons
/dashboard/system/basics/editor
/dashboard/system/basics/multilingual
/dashboard/system/basics/multilingual/update
/dashboard/system/basics/timezone
/dashboard/system/basics/reset_edit_mode

## Deprecated CSS class names ##

First buttons were using deprecated BS class names: pull-left, pull-right and btn-default so they were not aligned properly and not styled properly. All these class names were replaced with the new float-left, float-right and btn-secondary. Secondary buttons now have their border, and save buttons are on the right as should be. 

![action-bar](https://user-images.githubusercontent.com/1488833/81298838-def5c700-9064-11ea-83e4-89349e53d7fd.png)

The deprecated "hide" class name was used as well on the attribute edit page to show and hide a warning. So the hide/show functionality was not working. I replaced it with the proper d-none class name.

## Markup ##

I had to modify the markup here and there to follow the new design and make sure things were aligned properly. For instance some headings were using label instead of fieldset legend.

## Attribute dropdown & other issues ##

In the attribute page /dashboard/system/basics/attributes the attribute selection dropdown was not styled so I styled it.

![attribute-dropdown](https://user-images.githubusercontent.com/1488833/81298956-0ea4cf00-9065-11ea-9e57-99d833ced23d.png)

To make sure it looked as good as possible I also had to modify the FontAwesomeIconFormatter  class to make use of the fa-fw (fixed-width) class name. This ensures all icons use the same width so they are all properly aligned and the text on the right as well. For instance, the icon for the "User Group" attribute is normally wider than the others so the text was pushed further to the right.

Another problem was that the text and the data/time attributes were still using deprecated FontAwesome icons so they were not being displayed. I could see the FA 4 shim was used but it didn't seem to work so I replace the icon names wit their new FA 4 names in each attribute's controller.

## Success and error messages ##

Another problem was with success messages on single pages. Several of them were being set as "message" instead of "success" so they were blue instead of green. I fixed that and made sure error messages were displaying correctly.

I also had to replace the way error messages were set sometimes. For instance, I removed `$this->set('error', [$this->token->getErrorMessage()]);` and replaced it with `$this->error->add($this->token->getErrorMessage());`

## Editor preview ##

The editor preview on page /dashboard/system/basics/editor had a few problems.

First, after clicking on the preview button, which turned green, there was no indication of anything happening while the editor was loading. I added a loading text with a spinning FA 4 icon.

![editor-spinner](https://user-images.githubusercontent.com/1488833/81301721-f0d96900-9068-11ea-901a-cdab91697ae3.png)

Second, once the preview is active, checking extra plugins refreshes it. When clicking several checkboxes in rapid succession that was just too many server calls and the editor was flickering being reloaded on each click. So I used Underscore's _.debounce() function with a delay of 1500ms to only call the refresh once.
When clicking on checkboxes the loading messages is also displayed right away so the user knows something is happening.

I think both the loading indicator and the denounce make for much better user experience.

## Other fixes and additions ##

Form elements were given a 20px left padding but non-form elements were not aligned the same way. So I added a new class name ccm-dashboard-content-indented that adds a 20px left margin and can be used on dashboard pages to align non-form elements properly.

While working on the social icons page I noticed all 3 concrete5 social links were using HTTP instead of HTTPS. Not a great look. So I modified elemental's content.xml to fix that.

## Question ##

On the install languages page /dashboard/system/basics/multilingual/update, by default the list is not displayed. We have to click on the link at the top to show it.

In my opinion that makes for one needlessly really empty page and that link is not really a great indicator of what should be done here.

Couldn't we just display the list on page load and remove the hide/show functionality that's really not needed?

## Weird Stuff ##

It seems the page /dashboard/system/basics/images was removed but the single page and controller files are still there. Should I delete them?

It seems BootStrap collapse is broken all over. Try it on the Install languages page for instance /dashboard/system/basics/multilingual/update

This is the error I'm getting

```
main.js:4237 Uncaught TypeError: Cannot convert object to primitive value
    at RegExp.test (<anonymous>)
    at HTMLDivElement.<anonymous> (main.js:4237)
    at Function.each (jquery.js?ccm_nocache=2af556aaa1bab5ebc0174f18be041b6bf3d4575b:2)
    at S.fn.init.each (jquery.js?ccm_nocache=2af556aaa1bab5ebc0174f18be041b6bf3d4575b:2)
    at S.fn.init._jQueryInterface [as collapse] (main.js:4231)
    at HTMLDivElement.<anonymous> (main.js:4291)
    at Function.each (jquery.js?ccm_nocache=2af556aaa1bab5ebc0174f18be041b6bf3d4575b:2)
    at S.fn.init.each (jquery.js?ccm_nocache=2af556aaa1bab5ebc0174f18be041b6bf3d4575b:2)
    at HTMLElement.<anonymous> (main.js:4286)
    at HTMLDocument.dispatch (jquery.js?ccm_nocache=2af556aaa1bab5ebc0174f18be041b6bf3d4575b:2)
(anonymous) @ main.js:4237
each @ jquery.js?ccm_nocache=2af556aaa1bab5ebc0174f18be041b6bf3d4575b:2
each @ jquery.js?ccm_nocache=2af556aaa1bab5ebc0174f18be041b6bf3d4575b:2
_jQueryInterface @ main.js:4231
(anonymous) @ main.js:4291
each @ jquery.js?ccm_nocache=2af556aaa1bab5ebc0174f18be041b6bf3d4575b:2
each @ jquery.js?ccm_nocache=2af556aaa1bab5ebc0174f18be041b6bf3d4575b:2
(anonymous) @ main.js:4286
dispatch @ jquery.js?ccm_nocache=2af556aaa1bab5ebc0174f18be041b6bf3d4575b:2
v.handle @ jquery.js?ccm_nocache=2af556aaa1bab5ebc0174f18be041b6bf3d4575b:2
```

aembler commented:
Thanks @mnakalay - this is looking really really good. Just got a couple bits of feedback on this and then I'm sure we can get it merged. 

Name and Attributes, Accessibility, Social Links, Bookmark Icons all look good.

Rich Text Editor: I like the Preview button, it looks nice and seems to work well. However there's something off about the list of Editor plugins. The checkboxes seem misaligned and don't have enough padding between them and the labels. Also something about the tooltip is going crazy when I hover over it; check out this screenshot:

![editor_checkboxes](https://user-images.githubusercontent.com/527809/81325351-12c4f280-904d-11ea-8b34-9bb290a95e81.gif)

Mutilingual: The checkbox under Login on this page doesn't look aligned properly and doesn't have the correct padding next to the label.  The Update Languages page looks strange (see this screenshot): 

<img width="891" alt="image" src="https://user-images.githubusercontent.com/527809/81325571-659eaa00-904d-11ea-9cbc-1bd305eec912.png">

Time zone page looks great.

Reset edit mode page looks great. 

Thanks! Here are some answers to your additional questions:

1. Yes, we removed the images page - can you remove the controller and single page like you asked? That sounds like a good idea.
2. Yes, this is a new bug because we introduced jQuery 3.5, which apparently has a conflict with bootstrap 4. I expect a BS4 update will come soon to fix this issue (hopefully?)

mnakalay commented:
@aembler here are the changes and a few comments and interrogation.

Concerning the tooltips, there was a problem sitewide. The tooltip holder #ccm-tooltip-holder had `position: relative` set on it and all tooltips were misaligned everywhere. I removed it and it' working now.

There is a problem with checkboxes everywhere. There is no padding on them at all. When you see them with spacing between the checkbox and the label that's because there's a space in the markup not because of padding.
If we were to add padding we would have to make sure there is never any space in the markup between the checkbox and its text.
In the present situation, we have to make sure to add a space in the markup which is what I did for the editor page.
Really no perfect solution here.

Here's a screenshot of what it looks like now

![editor-alignment-tooltip](https://user-images.githubusercontent.com/1488833/81830894-2e883700-952c-11ea-88a7-0f41aa87ab2e.png)

Concerning the update languages page, I am not sure what the problem is? I added BootStrap classes to make it look better but that's all I could think of (screenshot below). What else should change?

![language-updates-badge](https://user-images.githubusercontent.com/1488833/81830958-45c72480-952c-11ea-8a58-aed4b7ed355c.png)

mlocati commented:
About the checkbox spacing question: I'd avoid forcing some padding via CSS but use spaces (it's more natural: do you want a space? Add a space)

mnakalay commented:
@mlocati I totally agree with you. It's better to say add a space when you need it rather than asking to make sure no extra space is added. Especially since indenting code with spaces creates spaces in the page anyway.

mnakalay commented:
Oups sorry about that. I'm on it

mlocati commented:
I did the same mistake in #8648 (fixed by https://github.com/concrete5/concrete5/pull/8648/commits/ad21639928f52cc83f02424537c1cc1801243013).

The problem with this new Bootstrap 4 syntax is that you have to know the value of the `id` attribute of the `<input>` element since you have to explicitly write it in the `for` attribute of the `<label>` element.

This may not be the case for `<input type="radio">`, where the value of the `id` attribute is dynamically build; BTW thanks to https://github.com/concrete5/concrete5/pull/8645 you can now explicitly define this value explicitly (see for example https://github.com/concrete5/concrete5/pull/8648/commits/ad21639928f52cc83f02424537c1cc1801243013#diff-8ef2aee1732d4b7b364ef44a6ce1f465R53

mnakalay commented:
@aembler I changed all the checkboxes to conform to BS4 rules.

mlocati commented:
> 2. Can you reopen a bedrock pull request with just your tooltip position relative changes?

I did it: https://github.com/concrete5/bedrock/pull/65

mnakalay commented:
Hi, @aembler Sorry I got sidetracked on another project that got time-sensitive out of the blue.
I fixed the issue.

@mlocati thank you for taking care of that.

mnakalay commented:
@aembler I'm sorry but I have absolutely no idea why all checks fail. A little help please.

aembler commented:
Now that we've moved over to the develop branch, can you reopen this against the develop branch?

mnakalay commented:
Yes I just saw your message in Slack about it. Will do shortly
